lib/node/runner/checker.go

```diff
@@ -731,13 +736,56 @@ func FinishTransactions(blk block.Block, transactions []*transaction.Transaction
 		if err = baSource.Save(st); err != nil {
 			return
 		}
-
 	}
-	return
+	return nil
+}
+
+//
+// Apply delayed operations for this block height
+//
+// If any operation is pending for this height, it will be applied by this function.
+// Since delayed operations are only added via previous operations,
+// the content of this pool is guaranteed to be in sync for any well-behaved node.
+//
+// Params:
+//   blk  = Pointer to the Block being finalized
+//   pool = The PendingPool to get operations from
+//   st   = Storage, used to look up the required blocks (and apply the changes)
+//
+// Returns:
+//   `nil` on success, in which case `pending` might have been mutated.
+//   If the return is non-`nil`, pending won't have been mutated.
+func finishDelayedOperations(blk *block.Block, pending *transaction.PendingPool, st *storage.LevelDBBackend) error {
+	var err error
+	offset := uint64(0)
+	for pendingH := pending.Peek(blk.Height, 0); pendingH != ""; pendingH = pending.Peek(blk.Height, offset) {
```

Comment:
I think, the `pendingHash` is better than `pendingH` because when I see this, I thought it was `pendingHeight`.

Reply:
Good point